File: Language/C#/Book/tmp/LINQ实战/Chapter06to08.CSharp/Common/SampleClasses/Ch7/AuthorXml.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Linq;
using System.Linq;
using System.Text;

namespace LinqInAction.Chapter06to08.Common.SampleClasses.Ch7
{
    public class AuthorXml
    {
        private System.Guid _Id;
        private EntitySet<BookAuthor> _BookAuthors = null;

        public System.Guid ID
        {
            get { return _Id; }
            set { _Id = value; }
        }
        public string LastName { get; set; }
        public string FirstName { get; set; }
        public string WebSite { get; set; }
        public byte[] TimeStamp { get; set; }

        public global::System.Data.Linq.EntitySet<BookAuthor> BookAuthors
        {
            get { return this._BookAuthors; }
            set { this._BookAuthors.Assign(value); }
        }
    }
}

File: Language/C#/Book/tmp/LINQ实战/Chapter06to08.VB/Common/SampleClasses/Ch7/AuthorXml.vb
Imports System.Linq
Imports System.Data.Linq

Namespace SampleClasses.Ch7
    Public Class AuthorXml
        Private _Id As Guid
        Private _BookAuthors As EntitySet(Of BookAuthor)

        Public Property ID() As System.Guid
            Get
                Return _Id
            End Get
            Set(ByVal value As System.Guid)
                _Id = value
            End Set
        End Property

        Private _LastName As String
        Public Property LastName() As String
            Get
                Return _LastName
            End Get
            Set(ByVal value As String)
                _LastName = value
            End Set
        End Property

        Private _FirstName As String
        Public Property FirstName() As String
            Get
                Return _FirstName
            End Get
            Set(ByVal value As String)
                _FirstName = value
            End Set
        End Property

        Private _WebSite As String
        Public Property WebSite() As String
            Get
                Return _WebSite
            End Get
            Set(ByVal value As String)
                _WebSite = value
            End Set
        End Property

        Private _TimeStamp As Byte()
        Public Property TimeStamp() As Byte()
            Get
                Return _TimeStamp
            End Get
            Set(ByVal value As Byte())
                _TimeStamp = value
            End Set
        End Property

        Public Property BookAuthors() As EntitySet(Of BookAuthor)
            Get
                Return _BookAuthors
            End Get
            Set(ByVal value As EntitySet(Of BookAuthor))
                _BookAuthors.Assign(value)
            End Set
        End Property
    End Class
End Namespace